### PR TITLE
Refactor self assessment screens

### PR DESCRIPTION
**Pull request by zhaccc**
#### Description:

- Refactor self-assessment components
- Separation of concerns on some reusable components
- Implement design changes

#### How to test:

Go to the self-assessment screen from the navbar.

### CODE REVIEW

File: app/components/Typography.tsx
Comment (johnschoeman) [inline app/components/Typography.tsx:8]: Are these all being used? it might be nice to move away from the `use` prop and just use the plain vanilla `style` prop provided by rn, so we dont have to maintain two separate patterns for styles.

**Review by jeffstolz — APPROVED** [review]
I think we should approve this and get it merged, although we'll want to do some refactoring to address the recent changes that have been applied to our designs and styling architecture. Rather than delay this PR further, I recommend we address them after merging.

File: app/views/assessment/EmergencyAssessment.js
Comment (johnschoeman) [inline app/views/assessment/EmergencyAssessment.js]: remove comment?

File: app/views/assessment/components/Button.js
Comment (johnschoeman) [inline app/views/assessment/components/Button.js:1]: is it possible to not have the assessment screen use a bespoke button component? could we be using the same button as the rest of the app?

File: app/views/assessment/endScreens/Caregiver.js
Comment (johnschoeman) [inline app/views/assessment/endScreens/Caregiver.js]: This MetaContext is pretty undescriptive it would be great to replace with something more focused in a future commit.

File: app/views/assessment/index.js
Comment (johnschoeman) [inline app/views/assessment/index.js:159]: Would it makes sense to have the navigator for the self assessment screen live with the other navigators? Or is there a good reason to keep it separate?

**Review by johnschoeman — APPROVED** [review]
Over all looks like a great step in the right direction. A few questions which i think would make more sense to capture in follow up prs. On that note, this is a pretty big PR in terms of changes, given how quickly this project moves and how many people are actively working on things it would be great to start breaking these down into smaller commits where possible.

File: app/views/assessment/components/Button.js
Comment (zhaccc) [reply app/views/assessment/components/Button.js:1]: There was a discussion about this and we decided not to use that button yet since it was refactored and added to too often. That being said I would like to open a different PR to implement the button we use throughout the app.

File: app/components/Typography.tsx
Comment (zhaccc) [reply app/components/Typography.tsx:8]: These are being used but I would really like to step away from those and use `style` prop as suggested.

File: app/views/assessment/endScreens/Caregiver.js
Comment (zhaccc) [reply app/views/assessment/endScreens/Caregiver.js]: Agreed, naming on this one is no good.

File: app/views/assessment/components/Button.js
Comment (johnschoeman) [reply app/views/assessment/components/Button.js:1]: makes sense to me!

File: app/views/assessment/index.js
Comment (zhaccc) [reply app/views/assessment/index.js:159]: Hmm, I am not sure yet because the file contains a lot of logic specific for the assessment. I do think we should be able to do this once we step away from the wey question flow is handled right now, and remove some of that logic.

**Pull request merged**
